src/main/java/com/google/cloud/teleport/io/WindowedFilenamePolicy.java

```diff
@@ -137,22 +378,70 @@ public ResourceId unwindowedFilename(
   /**
    * Resolves any date variables which exist in the output directory path. This allows for the
    * dynamically changing of the output location based on the window end time.
+   * A custom pattern can be used, if provided.
    *
    * @return The new output directory with all variables resolved.
    */
   private ResourceId resolveWithDateTemplates(
       ValueProvider<String> outputDirectoryStr, BoundedWindow window) {
     ResourceId outputDirectory = FileSystems.matchNewResource(outputDirectoryStr.get(), true);
-
+  
     if (window instanceof IntervalWindow) {
       IntervalWindow intervalWindow = (IntervalWindow) window;
       DateTime time = intervalWindow.end().toDateTime();
       String outputPath = outputDirectory.toString();
-      outputPath = outputPath.replace("YYYY", YEAR.print(time));
-      outputPath = outputPath.replace("MM", MONTH.print(time));
-      outputPath = outputPath.replace("DD", DAY.print(time));
-      outputPath = outputPath.replace("HH", HOUR.print(time));
-      outputPath = outputPath.replace("mm", MINUTE.print(time));
+  
+      // Default Date patterns.
+      String yearPattern = "YYYY";
+      String monthPattern = "MM";
+      String dayPattern = "DD";
+      String hourPattern = "HH";
+      String minutePattern = "mm";
+  
+      // Use a custom pattern, if provided.
+      if (year() != null) {
+        if (year().get() != null) {
+          yearPattern = year().get();
+        }
+      }
```

Comment:
Do you think it would be useful to log this at a debug level perhaps? At most we have a log every minute? See how this is done using MoreObjects.firstNotNull

https://github.com/apache/beam/blob/aa345dbeb9f6bdb36f5acf0fb8958777253acd51/sdks/java/io/splunk/src/main/java/org/apache/beam/sdk/io/splunk/SplunkEventWriter.java#L119-L129

Reply:
Yup, I'll add a few log statements for this